wsmaster/che-core-api-auth-github/src/main/java/org/eclipse/che/security/oauth/GitHubOAuthAuthenticator.java

```diff
@@ -24,17 +24,27 @@
 /** OAuth authentication for github account. */
 @Singleton
 public class GitHubOAuthAuthenticator extends OAuthAuthenticator {
+  private final String userRequestUrl;
+
   public GitHubOAuthAuthenticator(
-      String clientId, String clientSecret, String[] redirectUris, String authUri, String tokenUri)
+      String clientId,
+      String clientSecret,
+      String[] redirectUris,
+      String authEndpoint,
+      String authUri,
+      String tokenUri)
       throws IOException {
+    userRequestUrl =
+        isNullOrEmpty(authEndpoint) || authEndpoint.equals("https://github.com")
+            ? "https://api.github.com/user"
+            : authEndpoint + "/api/v3/user";
```

Comment:
I am afraid if `//` in the path can cause any issues

Reply:
Added `trimEnd()`